### PR TITLE
split train mode into pretrain & finetune

### PR DESCRIPTION
**Pull request by aobo-y**
--mode=pretrain
--mode=finetune

no more config update

### CODE REVIEW

**Comment by aobo-y** [comment]
@jiahao42 no, actually it used to work like this as well
if `n_iter` is less than `trained_iter`, it means training has finished

the iter in `config` means the maximum iteration we should train, so just continue from `trained_iter`

if we you wish to retrain from **1** to the maximum `iter` of config, you can call `trainer.reset()`
i called it when first time `finetune` with a `pretrain` checkpoint

File: src/chatbot.py [hunk 1]
```diff
@@ -129,17 +128,42 @@ def build_model(checkpoint_name):
...
+
+        corpus = config.FINETUNE_CORPUS
+        trim_corpus = False
+        train_persona = True
+        n_iter = config.FINETUNE_N_ITER
+
+        # finetune from pretrain checkpoint, reset start iter
+        # verify stage keyword to make it compatible with previous
+        if 'stage' not in checkpoint or checkpoint['stage'] != 'finetune':
+            trainer.reset_iter()
```
Comment (aobo-y) [inline src/chatbot.py]: @jiahao42 here, say we pretrained for 100K iteration, now I only want to `finetune` for another `10K` iteration with `pretrain` checkpoint, so i need to reset the `100K` iter to `1`

but if I `finetune` with a `finetune` checkpoint, just continue training. same for `pretrain` with `pretrain` checkpoint

Comment (aobo-y) [reply src/chatbot.py]: of course, one should never `pretrain` with `finetune` checkpoint......

Comment (jiahao42) [reply src/chatbot.py]: Oh sorry, I didn't see you will reset iter in fine tune here.
